core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2540,17 +2540,18 @@ private void mountInternal(Supplier<JournalContext> journalContext, LockedInodeP
               ExceptionMessage.UFS_PATH_DOES_NOT_EXIST.getMessage(ufsPath.getPath()));
         }
       }
-      // Check that the alluxioPath we're creating doesn't shadow a path in the default UFS
-      String defaultUfsPath = Configuration.get(PropertyKey.MASTER_MOUNT_TABLE_ROOT_UFS);
-      UnderFileSystem defaultUfs = UnderFileSystem.Factory.createForRoot();
-      String shadowPath = PathUtils.concatPath(defaultUfsPath, alluxioPath.getPath());
-      if (defaultUfs.exists(shadowPath)) {
-        throw new IOException(
-            ExceptionMessage.MOUNT_PATH_SHADOWS_DEFAULT_UFS.getMessage(alluxioPath));
+      // Check that the alluxioPath we're creating doesn't shadow a path in the parent UFS
```

Comment:
The logic here is that if there is a path /a/b/c where a,b,c are all mount points, if c shadows a, b must shadow a as well. so we would have done the check when we mounted b.  since b was successfully mounted, c can not shadow a, c can only shadow b. So we just have to check if c shadows b.